staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -1027,11 +1016,7 @@ func (o *ApplyOptions) PrintAndPrunePostProcessor() func() error {
 
 		if o.Prune {
 			if cmdutil.ApplySet.IsEnabled() && o.ApplySet != nil {
-				pruner, err := newApplySetPruner(o)
-				if err != nil {
-					return err
-				}
-				if err := pruner.pruneAll(ctx, o.ApplySet); err != nil {
+				if err := o.ApplySet.Prune(ctx, o); err != nil {
```

Comment:
Should `o.ApplySet.UpdateParent` be merged into `o.ApplySet.Prune` as well (and then UpdateParent made private)? IIRC I originally had it that way but moved it out because the parent update went in before pruning was implemented at all. I don't think there's a conceivable case where pruning was completed but parent update should not be attempted?